Marlin/pinsDebug.h

```diff
@@ -357,136 +364,192 @@ static void pwm_details(uint8_t pin) {
         }
-        if (!multi_name_pin && extended) pwm_details(pin);  // report PWM capabilities only on the first pass & only if doing an extended report
       }
       SERIAL_EOL();
     }  // end of IF
   } // end of for loop
 
   if (!found) {
-    sprintf(buffer, "PIN: %3d ", pin);
+    sprintf_P(buffer, PSTR("%sPIN: %3d "), start_string, pin);
     SERIAL_ECHO(buffer);
     print_port(pin);
     if (IS_ANALOG(pin)) {
-      sprintf(buffer, " (A%2d)  ", int(pin - analogInputToDigitalPin(0)));    // analog pin number
+      sprintf_P(buffer, PSTR(" (A%2d)  "), int(pin - analogInputToDigitalPin(0)));    // analog pin number
       SERIAL_ECHO(buffer);
     }
-    else
-      SERIAL_ECHO_SP(8);   // add padding if not an analog pin
-    SERIAL_ECHOPGM("<unused/unknown>");
-    if (get_pinMode(pin)) {
-      SERIAL_PROTOCOL_SP(12);
-      SERIAL_PROTOCOLPAIR("Output = ", digitalRead_mod(pin));
-    }
     else {
-      if (IS_ANALOG(pin)) {
-        sprintf(buffer, "   Analog in = %5d", analogRead(pin - analogInputToDigitalPin(0)));
-        SERIAL_ECHO(buffer);
+      SERIAL_ECHO_SP(8);   // add padding if not an analog pin
+      SERIAL_ECHOPGM("<unused/unknown>");
+      if (extended) {
+        #ifdef AVR_AT90USB1286_FAMILY  // Teensy IDEs don't know about these pins so must use FASTIO
+          if (pin == 46 || pin == 47) {
+            SERIAL_PROTOCOL_SP(12);
+            if (pin == 46) {
+              print_input_or_output(GET_OUTPUT(46));
+              SERIAL_PROTOCOL(READ(46));
+            }
+            else {
+              print_input_or_output(GET_OUTPUT(47));
+              SERIAL_PROTOCOL(READ(47));
+            }
+          }
+          else
+        #endif
+        {
+          if (get_pinMode(pin)) {
+            SERIAL_PROTOCOL_SP(12);
+            print_input_or_output(true);
+            SERIAL_PROTOCOL(digitalRead_mod(pin));
+          }
+          else {
+            if (IS_ANALOG(pin)) {
+              sprintf_P(buffer, PSTR("   Analog in = %5d"), analogRead(pin - analogInputToDigitalPin(0)));
+              SERIAL_ECHO(buffer);
+              SERIAL_ECHOPGM("   ");
+            }
+            else
+              SERIAL_ECHO_SP(12);   // add padding if not an analog pin
+
+            print_input_or_output(false);
+            SERIAL_PROTOCOL(digitalRead_mod(pin));
+          }
+          //if (!pwm_status(pin)) SERIAL_CHAR(' ');    // add padding if it's not a PWM pin
+          if (extended) pwm_details(pin);  // report PWM capabilities only if doing an extended report
+        }
       }
-      else
-        SERIAL_ECHO_SP(9);   // add padding if not an analog pin
-
-      SERIAL_PROTOCOLPAIR("   Input  = ", digitalRead_mod(pin));
+      SERIAL_EOL();
     }
-    //if (!pwm_status(pin)) SERIAL_CHAR(' ');    // add padding if it's not a PWM pin
-    if (extended) pwm_details(pin);  // report PWM capabilities only if doing an extended report
-    SERIAL_EOL();
```

Comment:
@Bob-the-Kuhn I had a little hiccup when merging, wasn't sure how the end of this function should be. Are these changes as they were?